gem/chhaajja/videos/models.py

```diff
@@ -0,0 +1,46 @@
+from django.db import models
+
+from wagtail.core.models import Page
+from wagtail.core.fields import RichTextField
+from wagtail.admin.edit_handlers import FieldPanel
+from wagtail.search import index
+
+class VideosIndexPage(Page):
+    intro = RichTextField(blank=True)
+
+    content_panels = Page.content_panels + [
+        FieldPanel('intro', classname="full")
+    ]
+
+    def get_context(self, request):
+        context = super().get_context(request)
+
+        # Add extra variables and return the updated context
+        context['videos_entries'] = VideosPage.objects.live()
+        context['latest_video'] = VideosPage.objects.latest('id')
+        return context
+
+
+
+
+# Keep the definition of BlogIndexPage, and add:
+
+
+class VideosPage(Page):
+	date = models.DateField("Post date")
+	intro = models.CharField(max_length=250)
+	body = RichTextField(blank=True)
+	videourl = models.CharField(max_length=250,null=True)
```

Comment:
wagtail has an embed module that handles media embeds automatically. Would it be easier to do video embeds that way? I see that the video url is used for an embed in the templates, if it is done this way, please provide some help text on the field around how the url should look, it is not clear that it should be a youtube url.

The benefit of using Wagtail embeds is that you are not restricted to using youtube. In future if you wanted to use any other media embed you would be able to do so without changing any code